writegood-mode.el

```diff
@@ -272,17 +278,104 @@ From Wikipedia URL `https://en.wikipedia.org/wiki/Flesch–Kincaid_readability_t
     ((and (<= 80.0 score) (< score 90.0)) "Easy (6th grade)")
     ((<= 90.0 score) "Very easy (5th grade)")))
 
-;;;###autoload
-(defun writegood-reading-ease (&optional start end)
-  "Flesch-Kincaid reading ease test in the region bounded by START and END.
+;;; BEGIN Score on the fly
+(defun writegood-calculate-reading-ease (&optional start end)
+  "Calculate score of Flesch-Kincaid reading ease test in the region bounded by START and END.
 
 Scores roughly between 0 and 100."
-   (interactive)
    (let* ((params (writegood-fk-parameters start end))
           (sentences (nth 0 params))
           (words     (nth 1 params))
           (syllables (nth 2 params))
           (score  (- 206.835 (* 1.015 (/ words sentences)) (* 84.6 (/ syllables words)))))
+     score))
+
+(defun writegoodmode-reading-ease-thing-at-point (thing)
+  "Calculate score for thing at point."
+  (let* ((bounds (bounds-of-thing-at-point thing))
+         (b (car bounds))
+         (e (cdr bounds)))
+    (if (and
+         (not (null b))
+         (not (null e))
+         ;; this is a guess: when the interval between boundaries is
+         ;; huge, the paragraph is too big to be validated.
+         (< (- e b) 100000))
+        (let ((score (writegood-calculate-reading-ease b e)))
+          (message "%s reading ease score: %.2f %s" (symbol-name thing) score
+            (writegood-reading-ease-score->comment score))))))
+
+;;;###autoload
+(defun writegoodmode-reading-ease-sentence ()
+  "Calculate score for the sentence at point."
+  (interactive)
+  (writegoodmode-reading-ease-thing-at-point 'sentence))
+
+;;;###autoload
+(defun writegoodmode-reading-ease-paragraph ()
+  "Calculate score for the paragraph at point."
+  (interactive)
+  (writegoodmode-reading-ease-thing-at-point 'paragraph))
+
+;;;###autoload
+(defun writegoodmode-reading-ease-page ()
+  "Calculate score for the page at point."
+  (interactive)
+  (writegoodmode-reading-ease-thing-at-point 'page))
+
+(defun writegood-after-sentence ()
+  "Calculate reading ease after a sentence is completed. Here we
+consider the sentence completion the addition of a dot."
+  (if (and (string-match-p (regexp-quote "self-insert-command") (symbol-name last-command)) (eq (char-before) 46)) (writegoodmode-reading-ease-sentence)))
```

Comment:
Is char 46 a '.'?  What if a sentence ends with a question-mark? Or exclamation point!?  We'll need a more general check for sentence end to apply here.

Reply:
You are perfectly right! I did not think about it at all.
Should I propose a set of characters that delimit a sentence or you have already a list in mind?
Maybe I can find something already done for `thing-at-point 'sentence`